### PR TITLE
Gradient placeholder component

### PR DESCRIPTION
**Pull request by jankrzempek**
**Ticket:** https://trello.com/c/IrXCCwaI/1723-value-representable-label
**Goal:** Scheme for the label which represents data from given sensor
**Use:** `<StationValueLabel date="12 Jun" value={1} parameter="PM2.5 µg/m"/>`

### CODE REVIEW

File: app/javascript/react/assets/styles/Typography/Typography.style.tsx [hunk 1]
```diff
@@ -1,34 +1,46 @@
-import styled, { css } from "styled-components";
+import styled from "styled-components";
+
+const LargeH1 = styled.h1`
+  font-family: "Roboto";
+  font-weight: 600;
+  font-size: 72px;
+  color: inherit
+`;
```
Comment (smialko) [inline app/javascript/react/assets/styles/Typography/Typography.style.tsx:8]: I'm not sure about the name. Those headers' main difference is font-size. Making one of them `largeH1` feels inconsistent. 🤔

File: app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.style.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import styled from "styled-components";
+
+const GradientBox = styled.div`
+  background: linear-gradient(
+      241deg,
+      rgba(0, 181, 239, 0.4) -2.4%,
+      rgba(0, 181, 239, 0) 94.94%
```
Comment (smialko) [inline app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.style.tsx:7]: Is there a possibility to make those `rgba(0, 181, 239, 0)` separate variables? Or move to Colors? Just to make it a little bit more readable :)

File: app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+const StationValueLabel = ({ date, value, parameter }: StationValues) => {
+  return (
+    <S.GradientBox>
+      <S.GradientImage>
+        <img src={BroadCastLogo} alt="Broadcasting logo" />
+      </S.GradientImage>
+      <S.TextContainer>
+        <H3>Avg for {date}</H3>
```
Comment (smialko) [inline app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.tsx:19]: Are we planning to have `Strings` file? If yes, why not start now :D

**Review by smialko — COMMENTED** [review]
👏 
you smashed it! keep reacting <🍞> 🤪

File: app/javascript/react/assets/styles/Typography/Typography.style.tsx [hunk 4]
```diff
@@ -1,34 +1,46 @@
-import styled, { css } from "styled-components";
+import styled from "styled-components";
+
+const LargeH1 = styled.h1`
```
Comment (franiakat) [inline app/javascript/react/assets/styles/Typography/Typography.style.tsx:3]: I see this component more as a span, that is just decorative, semantically it is not the most important information on the page, I think.
BTW there is always only one H1 element on the given page, I know right now we have none, though.

File: app/javascript/react/assets/styles/Typography/Typography.style.tsx [hunk 5]
```diff
@@ -37,6 +49,7 @@ const H6 = styled.h6<{ bold?: boolean; uppercase?: boolean }>`
   font-size: 10px;
   text-transform: ${(p) => (p.uppercase ? "uppercase" : "none")};
   letterspacing: ${(p) => (p.uppercase ? "0.1px" : "normal")};
+  color: inherit
```
Comment (franiakat) [inline app/javascript/react/assets/styles/Typography/Typography.style.tsx:52]: We can add a default color to the main tag like <body>, so all the text as a default will be the same and this inherit should not be needed.

Comment (franiakat) [reply app/javascript/react/assets/styles/Typography/Typography.style.tsx:52]: As well as Roboto font-family - should work.

File: app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.style.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    ),
+    #81dbcb;
```
Comment (franiakat) [inline app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.style.tsx:9]: Could you extract it to variable?

**Review by franiakat — COMMENTED** [review]
Happy to see your React PR! 💪 
I will continue this review tomorrow, I'll add more comments then, just for the sake of saving current work I am sending those.. :)

File: app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.tsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+interface StationValues {
+  date: string;
+  value: number;
+  parameter: string;
+}
+
+const StationValueLabel = ({ date, value, parameter }: StationValues) => {
+  return (
+    <S.GradientBox>
```
Comment (franiakat) [inline app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.tsx:14]: Personally I'd go with a naming along those lines:
```
    <S.Container>
      <S.ImageContainer>
        <img src={BroadCastLogo} alt="Broadcasting logo" />
      </S.ImageContainer>
      <S.TextContainer>
        <H3>Avg for {date}</H3>
        <LargeH1>{value}</LargeH1>
        <H3>{parameter}</H3>
      </S.TextContainer>
    </S.Container>
```

I see this gradient as sth irrelevant form the html perspective and it is enough that it is define as a css property.

File: app/javascript/react/components/molecules/StationValueLabel/index.ts [hunk 8]
```diff
@@ -0,0 +1 @@
+export * from "./StationValueLabel";
```
Comment (franiakat) [inline app/javascript/react/components/molecules/StationValueLabel/index.ts:1]: @jankrzempek  and @smialko do you have your VSCode set up the way it's adding an empty line after last line of code? If not, could you set it up. We could also agree for some VSCode common config in a file. If you are willing to prepare it, please do so in a separate PR. :)

File: app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.style.tsx [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+  flex-direction: column;
+  align-items: right;
+  text-align: right;
+  padding-right: 6%;
+  padding-top: 10%;
+  padding-bottom: 10%;
+  color: white;
+`;
+
+export { GradientBox, GradientImage, TextContainer };
```
Comment (franiakat) [inline app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.style.tsx:42]: I needed to refactor something and this is what I ended up with:

```
import styled from "styled-components";

const GradientBox = styled.div`
  background: linear-gradient(
      241deg,
      rgba(0, 181, 239, 0.4) -2.4%,
      rgba(0, 181, 239, 0) 94.94%
    ),
    #81dbcb;
  display: flex;
  flex-direction: row;
  align-items: flex-end;
  justify-content: space-between;
  padding: 34px 20px;
  width: 290px;
  height: 180px;
  border-radius: 10px;
`;

const GradientImage = styled.img`
  height: 115px;
  width: 115px;
`;

const TextContainer = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  text-align: right;
  color: white;
`;

const Label = styled.span`
  font-size: 16px;
  font-weight: 400;
`;

export { GradientBox, GradientImage, TextContainer, Label };
```
and then:
```
import React from "react";
import BroadCastLogo from "../../../assets/icons/broadCastLogo.svg";
import { LargeH1 } from "../../../assets/styles/Typography";
import * as S from "./StationValueLabel.style";

interface StationValues {
  date: string;
  value: number;
  parameter: string;
}

const StationValueLabel = ({ date, value, parameter }: StationValues) => {
  return (
    <S.GradientBox>
      <S.GradientImage
        src={BroadCastLogo}
        alt="Broadcasting logo"
      ></S.GradientImage>
      <S.TextContainer>
        <S.Label>Avg for {date}</S.Label>
        <LargeH1>{value}</LargeH1>
        <S.Label>{parameter}</S.Label>
      </S.TextContainer>
    </S.GradientBox>
  );
};

export { StationValueLabel };
```
Still this LargeH1 is to be changed as described elsewhere.
It will be hard to describe it, we can have a huddle and talk about the choices. :)

**Review by franiakat — CHANGES_REQUESTED** [review]
Please see the suggestions! 😎

File: app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.tsx (hunk 3 above)
Comment (smialko) [reply app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.tsx:19]: We used to make a separate file for all hard-coded Strings, for future localizing 🤔

Comment (franiakat) [reply app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.tsx:19]: Ok, thanks, I'll create the task for adding `i18n` to handle texts :)

File: app/javascript/react/assets/styles/Typography/Typography.style.tsx (hunk 5 above)
Comment (franiakat) [reply app/javascript/react/assets/styles/Typography/Typography.style.tsx:52]: Once it is fixed 😅

File: app/javascript/react/components/molecules/StationValueLabel/index.ts (hunk 8 above)
Comment (jankrzempek) [reply app/javascript/react/components/molecules/StationValueLabel/index.ts:1]: Setting up right now

File: app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.style.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+      241deg,
+      rgba(0, 181, 239, 0.4) -2.4%,
+      rgba(0, 181, 239, 0) 94.94%
+    ),
+    #81dbcb;
+  position: relative;
+  display: flex;
+  color: white;
+  width: 16%;
+  height: 11%;
```
[the remark this answers is not in the document]
Comment (jankrzempek) [reply app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.style.tsx:14]: How does it work? Isn't pixel value static? I thought that by using percentages we could have more generic approach

File: app/javascript/react/assets/styles/Typography/Typography.style.tsx (hunk 5 above)
Comment (jankrzempek) [reply app/javascript/react/assets/styles/Typography/Typography.style.tsx:52]: What do you mean by _to the main tag like_? How can it be done efficiently?

File: app/javascript/react/assets/styles/Typography/Typography.style.tsx (hunk 4 above)
Comment (jankrzempek) [reply app/javascript/react/assets/styles/Typography/Typography.style.tsx:3]: I
Zrzut ekranu 2023-11-23 o 20 19 09
 changed this to span, but then the spacing between elements is odd. Can you tell me why?

File: app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.style.tsx (hunk 9 above)
Comment (jankrzempek) [reply app/javascript/react/components/molecules/StationValueLabel/StationValueLabel.style.tsx:42]: Thank you!

**Review by franiakat — APPROVED** [review]
Looks great! 👏
